### PR TITLE
[ISSUE#1244]java sdk add nacos selector

### LINKED ISSUE #1244: [Feature] Java SDK adds nacos naming selecotor
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Feature Request

the sdk currently lacks name service addressing capabilities, so a selector module is added to support name service addressing. Currently implement nacos first.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by walterlife**
Fixes #1244  

### Motivation
the sdk currently lacks name service addressing capabilities, so a selector module is added to support name service addressing. Currently implement nacos first.

### Modifications
1. add nacos naming selector

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/selector/NacosSelector.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import com.alibaba.nacos.api.naming.pojo.Instance;
+
+public class NacosSelector implements Selector {
+    private static final Logger logger = LoggerFactory.getLogger(NacosSelector.class);
+    private NamingService namingService;
+
+    public void init() throws Exception {
+        try {
+            Properties properties = PropertiesUtils.readPropertiesFile(Constants.CONFIG_FILE_NAME);
+            namingService = NamingFactory.createNamingService(properties.getProperty(Constants.EVENTMESH_SELECTOR_NACOS_ADDRESS));
```
Comment (li-xiao-shuang) [inline eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/selector/NacosSelector.java:40]: Most production environments will enable authentication, so creating a NamingService requires adding username and password

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/utils/PropertiesUtils.java [hunk 2]
```diff
@@ -61,4 +62,14 @@ public static Properties getPropertiesByPrefix(final Properties from, final Prop
...
         return to;
     }
+
+    public static Properties readPropertiesFile(String fileName) {
+        try (final InputStream inputStream = PropertiesUtils.class.getClassLoader().getResourceAsStream(fileName)) {
+            Properties properties = new Properties();
+            properties.load(inputStream);
+            return properties;
+        } catch (Exception e) {
+            throw new IllegalArgumentException(String.format("File: %s is not exist", fileName));
```
Comment (li-xiao-shuang) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/utils/PropertiesUtils.java:72]: Is it inappropriate to throw an IllegalArgumentException here?

File: eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/selector/NacosSelector.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import com.alibaba.nacos.api.exception.NacosException;
+import com.alibaba.nacos.api.naming.NamingFactory;
+import com.alibaba.nacos.api.naming.NamingService;
+import com.alibaba.nacos.api.naming.pojo.Instance;
+
+public class NacosSelector implements Selector {
+    private static final Logger logger = LoggerFactory.getLogger(NacosSelector.class);
+    private NamingService namingService;
+
+    public void init() throws Exception {
```
Comment (li-xiao-shuang) [inline eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/selector/NacosSelector.java:37]: Whether to consider initializing only once

**Comment by li-xiao-shuang** [comment]
please check ci

File: eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/selector/NacosSelector.java (hunk 1 above)
Comment (walterlife) [reply eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/selector/NacosSelector.java]: done

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/utils/PropertiesUtils.java (hunk 2 above)
Comment (walterlife) [reply eventmesh-common/src/main/java/org/apache/eventmesh/common/utils/PropertiesUtils.java]: done

File: eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/selector/NacosSelector.java (hunk 3 above)
Comment (walterlife) [reply eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/selector/NacosSelector.java]: done add INIT_STATUS control variable

**Review by li-xiao-shuang — APPROVED** [review]

**Comment by li-xiao-shuang** [comment]
image

fix error

**Review by li-xiao-shuang — APPROVED** [review]

**Comment by li-xiao-shuang** [comment]
LGTM

**Pull request closed without merging**